internal/dispatch/keys/hasher_common.go

```diff
@@ -63,3 +66,17 @@ type hashableString string
 func (hs hashableString) AppendToHash(hasher hasherInterface) {
 	hasher.WriteString(string(hs))
 }
+
+type hashableContext struct{ *structpb.Struct }
+
+func (hc hashableContext) AppendToHash(hasher hasherInterface) {
+	jsonBytes, err := hc.MarshalJSON()
+	if err != nil {
+		// If the struct could not be marshaled, use the string form of the struct.
+		logging.Err(err).Msg("got error marshaling context")
+		hasher.WriteString(hc.String())
```

Comment:
If this were to happen, I fear it could somehow lead to a cache-key collision. Can we add some tests to exercise this? Shouldn't the request context have been validated in the API? if that's the case, maybe we should panic here as it would suggest an edge case?

Reply:
It won't, because the `String()` form of the struct is a proto text format: `fields:{key:"bar" value:{bool_value:true}} fields:{key:"foo" value:{number_value:1}}`

Reply:
Another option would be to panic

Reply:
yeah, panic is what I had in mind. Looks good 👍🏻